test/cmp_msg_test.c

```diff
@@ -31,6 +32,18 @@ typedef struct test_fixture {
     OSSL_CMP_PKISI *si;
 } CMP_MSG_TEST_FIXTURE;
 
+static OPENSSL_CTX *libctx = NULL;
+static OSSL_PROVIDER *default_null_provider = NULL, *provider = NULL;
+
+static X509 *X509_dup_with_libctx(const X509 *cert)
+{
+    X509 *dup = X509_dup(cert);
```

Comment:
Shouldnt cert already have  a libctx set?
X509_dup should be copying the libctx, propq if it doesnt already..

Reply:
>  X509_dup should be copying the libctx, propq if it doesnt already..

Yes, this is what one would expect, but as mentioned before in https://github.com/openssl/openssl/pull/11808#discussion_r471092440 this is not really the case, so I added this again as a workaround.
I confirmed this bug by adding `printf` to that function and using this test with and without the workaround.
I've just opened an issue on this bug: #12680.

Reply:
I would probably add something like
/* TODO(3.0) Clean this up - See issue #12680 */

Reply:
Done.